moto/ec2/models.py

```diff
@@ -3511,6 +3524,295 @@ def modify_subnet_attribute(self, subnet_id, attr_name, attr_value):
+        deliver_logs_permission_arn,
+    ):
+        if log_group_name is None and log_destination is None:
+            raise InvalidDependantParameterError(
+                "LogDestination", "LogGroupName", "not provided",
+            )
+
+        if log_destination_type == "s3":
+            if log_group_name is not None:
+                raise InvalidDependantParameterTypeError(
+                    "LogDestination", "cloud-watch-logs", "LogGroupName",
+                )
+        elif log_destination_type == "cloud-watch-logs":
+            if deliver_logs_permission_arn is None:
+                raise InvalidDependantParameterError(
+                    "DeliverLogsPermissionArn",
+                    "LogDestinationType",
+                    "cloud-watch-logs",
+                )
+
+        if max_aggregation_interval not in ["60", "600"]:
+            raise InvalidAggregationIntervalParameterError(
+                "Flow Log Max Aggregation Interval"
+            )
+
+    def create_flow_logs(
+        self,
+        resource_type,
+        resource_ids,
+        traffic_type,
+        deliver_logs_permission_arn,
+        log_destination_type,
+        log_destination,
+        log_group_name,
+        log_format,
+        max_aggregation_interval,
+    ):
+        # Guess it's best to put it here due to possible
+        # lack of them in the CloudFormation template
+        max_aggregation_interval = (
+            "600" if max_aggregation_interval is None else max_aggregation_interval
+        )
+        log_destination_type = (
+            "cloud-watch-logs" if log_destination_type is None else log_destination_type
+        )
+        log_format = (
+            "${version} ${account-id} ${interface-id} ${srcaddr} ${dstaddr} ${srcport} ${dstport} ${protocol} ${packets} ${bytes} ${start} ${end} ${action} ${log-status}"
+            if log_format is None
+            else log_format
+        )
+
+        # Validate the requests paremeters
+        self._validate_request(
+            log_group_name,
+            log_destination,
+            log_destination_type,
+            max_aggregation_interval,
+            deliver_logs_permission_arn,
+        )
+
+        flow_logs_set = []
+        unsuccessful = []
+
+        for resource_id in resource_ids:
+            deliver_logs_status = "SUCCESS"
+            deliver_logs_error_message = None
+            flow_log_id = random_flow_log_id()
+            if resource_type == "VPC":
+                # Validate VPCs exist
+                self.get_vpc(resource_id)
+            elif resource_type == "Subnet":
+                # Validate Subnets exist
+                self.get_subnet(resource_id)
+            elif resource_type == "NetworkInterface":
+                # Validate NetworkInterfaces exist
+                self.get_network_interface(resource_id)
+
+            if log_destination_type == "s3":
+                arn = log_destination.split(":", 5)[5]
```

Comment:
It hasn't been released yet, but we're introducing a dependency model where the user would only have to install the dependencies for the specific service he's interested in. 
(See the top of the Readme for installation instructions on what that would look like.)

The new `moto.logs` and `moto.s3` imports at the top would mean that everyone who uses EC2, would also have to install dependencies for Logs and S3.
Can we move the import statements inside the if/elif statement, so the import only happens when we absolutely need it? 
That way, only users who use S3 as a destination type would have to install dependencies for S3.

Reply:
Haven't thought that, it sounds smart, let me change that and thanks for catching that! :)